share/availability/full/availability.go

```diff
@@ -80,7 +87,15 @@ func (fa *ShareAvailability) SharesAvailable(ctx context.Context, root *share.Ro
 			return share.ErrNotAvailable
 		}
 	}
-	return err
+
+	if fa.store != nil {
+		err = fa.store.Put(ctx, root.Hash(), eds)
+		if err != nil && !errors.Is(err, dagstore.ErrShardExists) {
+			return fmt.Errorf("full availability: failed to store eds: %w", err)
+		}
+	}
```

Comment:
What is the case when `fa.store == nil`? It looks like misconfigured module, but I might miss something

Reply:
For testing etc

Reply:
There are no tests that utilises it :) If we don't use it, maybe remove the possibility for store operation to be skipped, because in this case we would not even notice it happened. 

Also nice to have would be to add unit test, covering storing logic.

Reply:
All full availability tests pass a nil store to NewShareAvailability via TestAvailability

Reply:
I've refactored the tests to include it and removed the nil checks inside of availability